### PR TITLE
A simple implementation of variables for the pharmacological expansion

### PR DESCRIPTION
**Pull request by gkarfakis19**
#### Summary
Features "Implements basic variables for the pharmacological expansion" 

#### Purpose of change

Adds some of the basic variables required to make #53152 work.

#### Describe the solution

As of yet I've implemented the code scaffolding required for the variables.
Additionally, I've used code from the existing heart_rate function to create some simple modifiers affecting heart rate.
Balancing WIP.

#### Describe alternatives you've considered

Not doing this.

#### Testing

Some basic testing was done to ensure all functions worked properly, using debugmsg() to display the values.

#### Additional context

Should the Heart Rate index calculations be as complicated? The original issue mentioned that these variables should initially be influenced solely by the consumption of drugs, but the presence of a well thought out heart_rate function in the code that depended on many factors made me reconsider this, as adapting it to work for heart rate index made sense. I'm especially looking for @I-am-Erk's thoughts on this.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
@Venera3 will want to look at this with me

**Comment by gkarfakis19** [comment]
How much do you think should fit into this single PR? As far I can see this can probably be merged after sorting out the heart rate modifiers, and separate PRs could be created with the harmful effects for low/high values of these variables as well as the effects of medication/drugs on them.

**Comment by fungamer2-2** [comment]
Seems interesting! This will definitely make things more realistic

File: src/character_body.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ static const efftype_id effect_infected( "infected" );
 static const efftype_id effect_mending( "mending" );
 static const efftype_id effect_narcosis( "narcosis" );
 static const efftype_id effect_sleep( "sleep" );
+static const efftype_id effect_cig( "cig" );
```
Comment (I-am-Erk) [inline src/character_body.cpp:45]: If I'm following that error right, this list should be in alphabetical order.

File: src/character_body.cpp [hunk 2]
```diff
@@ -1124,3 +1136,123 @@ bodypart_id Character::body_window( const std::string &menu_header,
...
+        temperature_modifier = 0.00333f;
+    }
+    if( has_trait( trait_COLDBLOOD3 ) || has_trait( trait_COLDBLOOD4 ) ) {
+        temperature_modifier = 0.005f;
+    }
+    const float hr_temp_mod = ( player_local_temp - 65 ) * temperature_modifier;
+    const float stamina_level = static_cast<float>( get_stamina() ) / get_stamina_max();
+    float hr_stamina_mod = 0.0f;
+    // The influence of stamina on heartrate seemeed excessive and was toned down.
+    if( stamina_level >= 0.9f ) {
```
Comment (I-am-Erk) [inline src/character_body.cpp:1165]: I don't like this as a nest of ifs. Let's just use an equation, the relationship can be mostly linear. I'm going to bed but I can think one up while I try to sleep.

Comment (I-am-Erk) [reply src/character_body.cpp:1165]:
```
hr_stamina_mod = 1.6 * ( 1 - stamina_level)
```
This creates more or less the same formula  you have here.

**Pull request merged**